haunted-houses/haunted-houses-service/src/main/java/cz/muni/fi/pa165/hauntedhouses/service/facade/AbilityFacadeImpl.java

```diff
@@ -3,25 +3,43 @@
 import cz.muni.fi.pa165.hauntedhouses.dto.AbilityCreateDTO;
 import cz.muni.fi.pa165.hauntedhouses.dto.AbilityDTO;
 import cz.muni.fi.pa165.hauntedhouses.facade.AbilityFacade;
+import cz.muni.fi.pa165.hauntedhouses.model.Ability;
+import cz.muni.fi.pa165.hauntedhouses.service.AbilityService;
+import cz.muni.fi.pa165.hauntedhouses.service.MappingService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
 
+import javax.transaction.Transactional;
+
+@Transactional
+@Service
 public class AbilityFacadeImpl implements AbilityFacade {
+
+    @Autowired
+    MappingService mappingService;
+
+    @Autowired
+    AbilityService abilityService;
+
     @Override
     public AbilityDTO findAbilityById(Long id) {
-        return null;
+        return mappingService.mapTo(abilityService.getAbilityById(id), AbilityDTO.class);
```

Comment:
Tady by mohla mappingService křičet v případě, že abilityServerice vrátí null, změnil bych to na
`Ability ability = abilityService.getAbilityById(id); `
`return (ability == null) ? null : mappingService.mapTo(ability, AbilityDTO.class);`